### DISCUSSION TITLE
Automations - would like the ability to run custom queries (instead of just querying rows of a single table)

### DISCUSSION THREAD

**Discussion by proxy318** [Budibase/budibase#9418] [Help] [answered]
I have a staff info database and record changes to staff details in a table. Each night, I have an email that sends out the changes from that day. Since it's a relational db, only the staff id is recorded in the log table. It would be nice if I could use a query to join the log table to the staff table, and in email, show the details (name, email, etc) of the staff member that changed. The query rows function of the automations only allow me to include the primary display of the staff table, and so is limiting.

**Comment by melohagan** [comment, marked as answer]
Hey @proxy318 

You can do this with the **External Data Connector**.

![Screenshot 2023-01-25 at 11 44 08](https://user-images.githubusercontent.com/101575380/214555027-504a859a-3d76-4bc2-b543-db8f5be86972.png)

![Screenshot 2023-01-25 at 11 44 25](https://user-images.githubusercontent.com/101575380/214555077-03b36a9e-310c-438d-854e-97340c731dd7.png)

In your case you can any complex query you like.

**Reply by proxy318** [reply]
That's awesome, thank you!
